### PR TITLE
Remove legacy code

### PR DESCRIPTION
**Pull request by danielgtaylor**
This PR does the following:
- [x] Remove `legacy` folder and associated tests (in favor of [metamorphoses](https://github.com/apiaryio/metamorphoses))
- [x] Simplify the build system and use `peasant`

This is part of #51.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,9 @@ API Description SDK
 > _Wardaddy: [Best job I ever had](http://www.imdb.com/title/tt2713180/quotes?item=qt2267083)._
 
 Fury provides uniform interface to API description formats such as
-[API Blueprint][].
+[API Blueprint][http://apiblueprint.org] and [Swagger](http://swagger.io/).
```
Comment (pksunkara) [inline README.md]: `[API Blueprint](https://apiblueprint.org)`

**Comment by pksunkara** [comment]
Can you divide the PR into 2 parts? One for all the non-legacy-removal things, and the other for only legacy-removal stuff.

That way, we can merge the first PR into master and then use that as a base for shared/legacy branch.

**Comment by danielgtaylor** [comment]
@pksunkara you can see most of this PR split out in #53. Once that gets merged in, this PR will have a much smaller diff that just removes the legacy code and switches to the new build system.

**Comment by pksunkara** [comment]
Can you update the PR and it's description? Thanks.

**Comment by pksunkara** [comment]
Also, couldn't we have merged a few of the eslint and package.json changes in the recent PR?

**Comment by danielgtaylor** [comment]
@pksunkara the PR and description are updated! I didn't include the ESLint and `package.json` changes in the other PR because they depend on using Peasant, and that in turn can't handle Coffeescript (which the legacy code was written in).

**Comment by kylef** [comment]
:+1: Looks good to me.

**Comment by pksunkara** [comment]
:+1: lgtm.

@danielgtaylor Can this be merged? What is the status of metamorphoses?

**Comment by danielgtaylor** [comment]
@kylef @pksunkara core is switching to Metamorphoses this sprint, so in my opinion we should probably wait until that is complete.

**Pull request merged**